### ISSUE TITLE
Can't pass the expected timezone in my test

### ISSUE THREAD

**Issue by idaneliz** [Jimbly/timezone-mock#1]
I trying to use your timezone mock npm package for to simulate another timezone on the Node JS in order to avoid a gap between the Linux and the timezone in my test case. 
But I can't pass the tz mock param and the test is failed before the script is run.

**This is the test case -**

```
describe('New Button test cases', function() {

    var EC = protractor.ExpectedConditions;
    var assert = require('assert');
    var timezone_mock = require('timezone-mock');
    function buggyCode() {
      // This function is potentially a bug since it's interpreting a string in 
      // the local timezone, which will behave differently depending on which 
      // system it is ran on. 
      return new Date('2015-01-01 12:00:00').getTime();
    }
    var result_local = buggyCode();
    timezone_mock.register('US/Pacific');
    var result_pacific = buggyCode();
    timezone_mock.register('US/Pacific');
    var result_eastern = buggyCode();
    assert.equal(result_local, result_pacific); // Might fail 
    assert.equal(result_pacific, result_eastern); // Definitely fails 


   it('New Button - New appointment dialog - Selecet service & Time', function() {
      buggyCode();
      editLink.isDisplayed().then(function (isVisible) {
        if(isVisible){
          console.log("-=- EDIT LINK IS VISIBILE -=- ");
          browser.wait(EC.visibilityOf(editLink), 30000);
          editLink.click();
          browser.wait(EC.visibilityOf(element(by.model("newAppointment.padding"))), 30000);
          // element(by.model("newAppointment.padding")).click();
          // browser.wait(EC.visibilityOf(element(by.xpath("//div[9]/md-select-menu/md-content/md-option[1]"))), 30000);
          // RandomPerparation.click();
          // browser.waitForAngular();
        } else {
          console.log("-=- EDIT LINK IS  -NOT- VISIBILE -=- ");
          browser.wait(EC.visibilityOf(element(by.model("newAppointment.padding"))), 30000);
          // element(by.model("newAppointment.padding")).click();
          // browser.wait(EC.visibilityOf(element(by.xpath("//div[9]/md-select-menu/md-content/md-option[1]"))), 15000);
          // RandomPerparation.click();
          browser.waitForAngular();
        }
      });
    });
```

**This is the error -**

```
[32m login - Forget password screen - Invalid email address - pass[0m
Waiting for an email...
/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/timezone-mock/index.js:28
        assert.ok(false, 'Unhandled date format passed to MockDate constructor: ' + param);
               ^
AssertionError: Unhandled date format passed to MockDate constructor: 29-May-2016 11:06:31 +0000
    at new MockDate (/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/timezone-mock/index.js:28:16)
    at Parser.parseFetch (/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/imap/lib/Parser.js:436:13)
    at Parser._resUntagged (/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/imap/lib/Parser.js:257:24)
    at Parser._parse (/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/imap/lib/Parser.js:137:16)
    at Parser._tryread (/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/imap/lib/Parser.js:82:15)
    at TLSSocket.Parser._cbReadable (/var/lib/jenkins/workspace/Sanity_Testing_node_12_2/node_modules/imap/lib/Parser.js:53:12)
    at TLSSocket.emit (events.js:104:17)
    at emitReadable_ (_stream_readable.js:424:10)
    at emitReadable (_stream_readable.js:418:7)
    at readableAddChunk (_stream_readable.js:174:11)
[launcher] Process exited with error code 1
[31m>> [39m
[33mWarning: Tests failed, protractor exited with code: 1 Use --force to continue.[39m

[31mAborted due to warnings.[39m
```

AS you can see the error receives before another script.. 
Something wired.

Please your Help,

Thanks.

**Comment by Jimbly** [comment]
Hi!

The error seems to indicate a date is getting converted to a string (not with a mock date, but with the Javasript Date in a European timezone), and then passed back to a constructor of a MockDate.  I think you might want to use all mocked or nothing, or perhaps it's finding what is actually a problem in the tested code - someone is passing a local-formatted date string to a constructor.  That being said, that date string appears to be unambiguous (since it has a timezone adjust), so, if it's actually part of the spec of Dates Javascript is supposed to handle, then adding an additional if clause to https://github.com/Jimbly/timezone-mock/blob/master/index.js#L27 would be the correct fix.  I'd take a PR for that, or if you have a branch somewhere that this can be reproduced on (although perhaps won't reproduce on my US-formatted dates with my current OS settings...), I could take a look.

**Comment by Jimbly** [comment]
Actually, it would just be a new RegEx to https://github.com/Jimbly/timezone-mock/blob/master/index.js#L22 - if this is a valid, unambiguous date string (looks like it's a valid RFC 2822 Date string), MockDate should parse it... try this branch/PR and see if it fixes the issue for you: https://github.com/Jimbly/timezone-mock/pull/2 (completely untested ; )

**Comment by Jimbly** [comment]
I never heard anything back, but was in this code, so added some tests, and that PR seems like it should fix your issue.  This has been merged into master and published to NPM as version 0.0.3.
